plan/logical_plan_test.go

```diff
@@ -1362,3 +1363,165 @@ func (s *testPlanSuite) TestAggPrune(c *C) {
 		c.Assert(ToString(p), Equals, ca.best, comment)
 	}
 }
+
+func (s *testPlanSuite) TestVisitInfo(c *C) {
+	defer testleak.AfterTest(c)()
+	cases := []struct {
+		sql string
+		ans []visitInfo
+	}{
+		{
+			sql: "insert into t values (1)",
+			ans: []visitInfo{
+				{mysql.InsertPriv, "test", "t", ""},
+			},
+		},
+		{
+			sql: "delete from t where a = 1",
+			ans: []visitInfo{
+				{mysql.DeletePriv, "test", "t", ""},
+				{mysql.SelectPriv, "test", "t", ""},
+			},
+		},
+		{
+			sql: "delete from a1 using t as a1 inner join t as a2 where a1.a = a2.a",
```

Comment:
you should add a multi join case like 'delete from t1, t2, t3, t4'. I think you should use different tables.